src/core/utils/compatibility.js

```diff
@@ -142,22 +179,42 @@ export function isCompatibleWithUserAgent({
   return { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX };
 }
 
+export type GetClientCompatibilityParams = {|
+  _log?: typeof log,
+  _window?: typeof window | {},
+  addon: AddonType,
+  clientApp: string,
+  currentVersion: AddonVersionType,
+  userAgentInfo: UserAgentInfoType,
+|};
+
+export type ClientCompatibilityType = {|
+  compatible: boolean,
+  downloadUrl?: string,
```

Comment:
Changing it to `string | null` would require also changing the consumer, which I'd rather not have to do in this patch, so I've changed it to `string | void`. Just wondering, why is `downloadUrl: string | void` considered better than `downloadUrl?: string`?

Reply:
ok, that's fine. For the case of a return value object it's not such a big deal but it's a big deal for parameter objects because `downloadUrl?` would allow the caller to forget to pass in the parameter. This could be a mistake. We want Flow to catch those mistakes but still allow the caller to say `someCall({ downloadUrl: undefined })`.

Reply:
Just following up on this because I still don't fully understand (and I commented on some `?`s in https://github.com/mozilla/addons-frontend/pull/6865). Are you saying that we should *never* have optional arguments? That every caller should have to pass either `null` or `undefined` explicitly into every function, even if the argument is irrelevant to the caller?